src/main/java/com/amazon/opendistroforelasticsearch/jobscheduler/scheduler/JobScheduler.java

```diff
@@ -158,7 +162,20 @@ boolean reschedule(ScheduledJobParameter jobParameter, JobSchedulingInfo jobInfo
             JobExecutionContext context = new JobExecutionContext();
             context.setExpectedExecutionTime(jobInfo.getExpectedPreviousExecutionTime());
             context.setJobVersion(version);
-            jobRunner.runJob(jobParameter, context);
+
+            if (jobParameter.getLockDurationSeconds() == null) {
+                jobRunner.runJob(jobParameter, context);
+            } else {
+                final LockModel lock =
+                    lockService.acquireLock(jobInfo.getIndexName(), jobInfo.getJobId(), jobParameter.getLockDurationSeconds());
```

Comment:
Providing index name as jobType in acquireLock, if using indexName let's switch the parameter to indexName?